rest/client/client.go

```diff
@@ -51,16 +51,21 @@ type TaskData struct {
 // the API server. To change the default retry behavior, use the SetTimeoutStart, SetTimeoutMax,
 // and SetMaxAttempts methods.
 func NewCommunicator(serverURL string) Communicator {
-	return &communicatorImpl{
+	c := &communicatorImpl{
 		maxAttempts:  defaultMaxAttempts,
 		timeoutStart: defaultTimeoutStart,
 		timeoutMax:   defaultTimeoutMax,
 		serverURL:    serverURL,
 		httpClient:   util.GetHttpClient(),
 	}
+	c.httpClient.Timeout = heartbeatTimeout
+	return c
 }
 
-func (c *communicatorImpl) Close() { util.PutHttpClient(c.httpClient) }
+func (c *communicatorImpl) Close() {
+	c.httpClient.Timeout = time.Duration()
```

Comment:
The init function in http_client_pool.go sets `Timeout` to `5 * time.Minute`. Should there maybe be a reset function called by `PutHttpClient` so that we don't rely on clients to reset this to the right value?